wdl/Module04b.wdl

```diff
@@ -558,9 +559,32 @@ task GetMedianSubset{
     max_retries: 1
   }
   RuntimeAttr runtime_attr = select_first([runtime_attr_override, default_attr])
-  command {
-    R -e 'd<-read.table("~{medians}",header=T,check.names=F);result<-d[((apply(as.matrix(d[,5:length(d)]),FUN=median,1)<0.97 & apply(as.matrix(d[,5:length(d)]),FUN=median,1)>0.85)|(apply(as.matrix(d[,5:length(d)]),FUN=median,1)>1.03 & apply(as.matrix(d[,5:length(d)]),FUN=median,1)<1.65)),1:4];write.table(result,file="~{batch}_to_regeno.bed",sep="\t",quote=F,row.names=F,col.names=F)'
-  }
+  command <<<
+    set -euo pipefail
+    python3 <<CODE
+
+    from statistics import median
+    with open("~{medians}", 'r') as inp, open("~{batch}_to_regeno.bed", 'w') as outp:
+      for line in inp:
+        fields = line.strip().split('\t')
+        # first 4 fields are variant info (chr, start, end, varID)
+        var_info = fields[0:4]
+        # next len-4 fields are sample coverage medians for each sample in the batch
+        try:
+          coverage_medians = [float(x) for x in fields[4:]]
+        except ValueError as e:
+          if str(e) == "could not convert string to float: '.'":
+            # if missing data, print variant ID and chrom:start-end in case follow-up desired
+            print("Empty coverage data for variant %s at %s:%s-%s" % (var_info[3], var_info[0], var_info[1], var_info[2]))
```

Comment:
I'm slightly concerned this could generate a lot of unnecessary stdout that would end up in the execution bucket.